ui/components/multichain/pages/send/components/quote-card/index.tsx

```diff
@@ -118,6 +118,33 @@ export function QuoteCard({ scrollRef }: QuoteCardProps) {
     return () => clearTimeout(timeout);
   }, [timeLeft]);
 
+  // use to track when a quote is requested and received
+  useEffect(() => {
+    if (isSwapQuoteLoading) {
+      trackEvent(
+        {
+          event: MetaMetricsEventName.sendSwapQuoteRequested,
+          category: MetaMetricsEventCategory.Send,
+          sensitiveProperties: {
+            ...sendAnalytics,
+          },
+        },
+        { excludeMetaMetricsId: false },
+      );
+    } else {
+      trackEvent(
```

Comment:
This block can prematurely publish `sendSwapQuoteReceived` when the Send page is initially loaded so I think we need to check if `bestQuote` is defined:

```suggestion
    } else if (bestQuote) {
```

Reply:
good catch. I've flipped the blocks and added an extra condition to the first conditional.

Reply:
Can you also add a condition to the else block so it's only evaluated when quotes are loading? Otherwise `sendSwapQuoteRequested` will be emitted when there are no quotes received or if there's a fetch error